### DISCUSSION TITLE
help me :)  sorry guys for my first time

### DISCUSSION THREAD

**Discussion by ghost** [community/community#34759] [New to GitHub] [answered]
hi guys im new from this fantastic site!! nice to meet to all!!
i want a information about this
https://github.com/vichan-devel/vichan
i want to install this ! because i really love the imageboard! can someone tell me how i can install? because this is my first time !
(if you want to explain me because im a little noob) xD hahah sorry for everthing and have a nice day !!!

**Comment by airtower-luna** [comment]
Scroll down a little on the repository page to find the README. I see there are install instructions in there. :wink: 

If you have trouble with those instructions, posting your question in the discussions tab *of that repository* might work better because the developer(s) are much more likely to see it than here in the general community.

**Comment by DSDmark** [comment]
Hey,
Every repo on GitHub provide the instruction of install or use (software or packages). That all instruction mention the particular file called readme.md.
in your case, it also provides installing instruction as @airtower-luna mention like this:-

![Screenshot from 2022-10-01 10-48-45](https://user-images.githubusercontent.com/85744816/193393968-d219a66b-1b65-428c-b169-f8f0d93749bf.png)

## here is the above instruction :-

### Download and extract vichan to your web directory or get the latest development version with:

> git clone git://github.com/vichan-devel/vichan.git

> run composer install

 inside the directory

Navigate to install.php in your web browser and follow the prompts.

> vichan should now be installed. Log in to mod.php with the default username and password combination: admin / password.

Please remember to change the administrator account password.

**Reply by ghost** [reply]
hi thanks for respond me but i did ! nothing say HTTP ERROR 500  i can´t ( i understand for you is reall easy but for me is my first time is a hell!!)

**Reply by ghost** [reply]
look i extracted on my cpanel like this [https://ibb.co/8r91dpH](url) and when i try to install say me "cannot process this request at this time"
pls help me guys :(

**Reply by DSDmark** [reply]
Sorry but I bit confuse you already install that software, or you try to install then that error pops up  
> "cannot process this request at this time"

in second case, if you have problem  to installation, and you install that software like that
> git clone git://github.com/vichan-devel/vichan.git

and is not work, then you can download that software like 
![Screenshot from 2022-10-02 10-35-37](https://user-images.githubusercontent.com/85744816/193438891-94dd3639-89d8-4057-b882-c387542c54a2.png)

In above image there are download zip option.

**Reply by ghost** [reply]
sorry for the replay late but i already extracted the file in cpanel, but when i going to open "mysite"install.php doesnt work!! and if you want (i know im a noob xD ahhaha) can explain me with more details on install because i don't understand nothing

**Reply by ghost** [reply]
cpanel file manager like this and i can't install
![work](https://user-images.githubusercontent.com/107584070/193471557-18809ac3-2730-49de-b332-a374aa5dc9a3.PNG)

**Reply by ghost** [reply]
![help](https://user-images.githubusercontent.com/107584070/193471966-7b5e441e-db35-4e1d-a9d0-d5cf3089769d.PNG)
and show me this ... i really i dont know

**Reply by ghost** [reply]
and for the last what means run composer? because in my host i don't have  sorry guys for many request but i really want to learn to install!!!

**Comment by airtower-luna** [comment]
Composer is a package manager for PHP. I'm not too familiar with PHP, but it should retrieve and install dependencies (other libraries/tool) that the software you want to install needs.

**Reply by ghost** [reply]
airtower-luna ty for the responds me but if you see the sreenshot i have problem with installation 
i really i don't know how to install... if you want (only if you want)  can you help me with step by step (for noob like me) as install this vychan?

**Reply by airtower-luna** [reply]
I don't see any attempt at installing in your screenshot, only some sort of file browser. From the instructions I can tell you'll need to install [Composer](https://github.com/composer/composer) first so you can run the `composer install` command, but I don't know enough about PHP to help beyond that.

**Comment by DSDmark** [comment]
Sorry but i also not good with PHP, but I try to help if it is ok with you
first thing :- you need to download that repo in your local system.

> https://github.com/vichan-devel/vichan

Do not extract on direct in cPanel, first you need to extract in local system of your. 

Then run that comment :-

``` composer install ```

if you use Linux then you need to install it first 

``` sudo apt install composer ```

i not familiar with window but for window installation 
read this:- 

> https://www.javatpoint.com/how-to-install-composer-on-windows

after composer, you're ready to upload a file or folder to cPanel

but I guess it still not work so,

add your database. Open your SQL database and in your folder there is a file which name is **install.sql**
upload or add that file to your database 

try that all step and I guess it may going to work 👍🏻

**Comment by ghost** [comment]
guys ty :) i really love you :) this community is fantastic!!! 
![JAAAHHH](https://user-images.githubusercontent.com/107584070/193663516-6dd7b1d5-959c-44fa-b9c7-aca836c4c3b3.PNG)
now i nee to settings but will see how to continue :) ty guys

**Comment by ghost** [comment, marked as answer]
sorry guys the last question why when i finish the install the imageboard i any time when i going to my url appear this? i
![ok](https://user-images.githubusercontent.com/107584070/193667833-2b039992-54fb-4eca-ac48-ca71f1c9de73.PNG)

**Reply by ghost** [reply]
i already installed the imageboard but i really Don't Understand why my url appear this list?

**Reply by airtower-luna** [reply]
It looks like your webserver doesn't know it should run `index.php` for accesses to the directory instead of showing a directory listing. How to configure that depends on your webserver.

**Reply by DSDmark** [reply]
In that case, you need to Create a ***.htaccess** file in your root folder and add this line of code 

```
DirectoryIndex index.php index.html

RewriteEngine On

RewriteRule !^SourceFolder/ http://localhost:80/vichan-master/install.php [L,NC]
 ```

hey are the things you need to notice ```RewriteRule !^SourceFolder/ http://localhost/vichan-master/{yourfilepath} [L,NC]```

if you got your answer, then you need to close this question by **click** on the **accept** button. 
which answer you like :-

**Reply by ghost** [reply]
i did but not work because i dont have both index.php and index.html ... this is really a nightmare now can finally understand how install!!

**Reply by DSDmark** [reply]
Which file you want to open, just give the path of file here 

for example:-

1:
 ```
DirectoryIndex install.php 

RewriteEngine On

RewriteRule !^SourceFolder/ http://localhost:7000/vichan-master/install.php [L,NC]

```

2: 
```
DirectoryIndex index.html 

RewriteEngine On

RewriteRule !^SourceFolder/ http://localhost/vichan-master/templates/ [L,NC]

```

**Reply by DSDmark** [reply]
if you got your answer, then you should to close this question by click on the **accept answer** button.

**Comment by ossanna16** [comment]
Thank you for asking a great question @KenmochiShiena, I'm so glad you found our community 🙂
